File: flutter/lib/models/book.dart
class BookModel {
  final int id;
  final String title;
  final String author;
  final String isbn;
  final double price;
  final int stockQuantity;
  final String description;
  final String category;
  final String coverImage;

  BookModel({
    required this.id,
    required this.title,
    required this.author,
    required this.isbn,
    required this.price,
    required this.stockQuantity,
    required this.description,
    required this.category,
    required this.coverImage,
  });

  factory BookModel.fromJson(Map<String, dynamic> json) {
    return BookModel(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      author: json['author'] ?? '',
      isbn: json['isbn'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      stockQuantity: json['stock_quantity'] ?? 0,
      description: json['description'] ?? '',
      category: json['category'] ?? '',
      coverImage: json['cover_image'] ?? '',
    );
  }
}

File: core_app/app/models/book.py
from .. import db
from datetime import datetime

class Book(db.Model):
    __tablename__ = 'books'

    id = db.Column(db.Integer, primary_key=True)
    title = db.Column(db.String(200), nullable=False)
    author = db.Column(db.String(100), nullable=False)
    isbn = db.Column(db.String(20), unique=True, nullable=False)
    price = db.Column(db.Float, nullable=False)
    stock_quantity = db.Column(db.Integer, default=0)
    description = db.Column(db.Text, nullable=True)
    category = db.Column(db.String(50), nullable=True)
    cover_image = db.Column(db.String(200), nullable=True)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)

    orders = db.relationship('Order', backref='book', lazy=True)
